node/risk-app/server/src/js/map-page/map-page.js

```diff
@@ -0,0 +1,251 @@
+        return
+      }
+
+      for (let j = 0; j < category.maps.length; j++) {
+        map = category.maps[j]
+        if (i === 0 && j === 0) {
+          defaultMap = map
+        }
+
+        if (map.ref === ref) {
+          this.currMap = map
+          this.currCategory = category
+          return
+        }
+      }
+    }
+
+    this.currMap = defaultMap
+    this.currCategory = defaultCategory
+  }
+}
+
+function mapPage () {
+  getInitialKeyOptions()
+
+  function getParameterByName (name) {
+    name = name.replace(/[[]/, '\\[').replace(/\]/, '\\]')
+    const regex = new RegExp(`[\\?&]${name}=([^&#]*)`)
+    const results = regex.exec(window.location.search)
+    return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '))
+  }
+  const measurements = document.querySelectorAll('.govuk-radios__inputs')
+  const map = document.getElementById('map')
+  const body = document.body
+
+  const easting = parseInt(getParameterByName('easting'), 10)
+  const northing = parseInt(getParameterByName('northing'), 10)
+  const hasLocation = !!easting
+
+  mapPageConsts.maps.loadMap((hasLocation && [easting, northing]))
+
+  // This function updates the map to the radio button you select (extent, depth, velocity)
+  function setCurrent (ref) {
+    const mapController = new MapController(mapCategories.categories)
+    mapController.setCurrent(ref)
+    const selectedAddressCheckbox = document.getElementById('selected-address-checkbox')
+    const showFloodingCheckbox = document.getElementById('display-layers-checkbox')
+    const mapReferenceValue = selectedOption()
+
+    if (showFloodingCheckbox.checked) {
+      mapPageConsts.maps.showMap(`risk:${mapReferenceValue.substring(mapReferenceValue.indexOf('_') + 1)}`, selectedAddressCheckbox.checked)
+    } else {
+      mapPageConsts.maps.showMap(`risk:${mapReferenceValue.substring(mapReferenceValue.indexOf('_') + 1)}DONOTDISPLAY`, selectedAddressCheckbox.checked)
+    }
+  }
+
+  // Default to the first category/map
+  mapPageConsts.maps.onReady(function () {
+    measurements.forEach(function (measurement) {
+      if (measurement.name === 'measurements') {
+        measurement.addEventListener('change', function (event) {
+          event.preventDefault()
+          setCurrent(event.target.value)
+        })
+      }
+      if (measurement.name === 'scenarios-depth') {
+        measurement.addEventListener('change', function (event) {
+          event.preventDefault()
+          setCurrent(event.target.value)
+        })
+      }
+      if (measurement.name === 'scenarios-velocity') {
+        measurement.addEventListener('change', function (event) {
+          event.preventDefault()
+          setCurrent(event.target.value)
+        })
+      }
+      if (measurement.name === 'map-toggle') {
+        measurement.addEventListener('change', function (event) {
+          event.preventDefault()
+          setCurrent(event.target.value)
+        })
+      }
```

Comment:
A lot of duplication here. The event listener function could be a variable, or the multiple if statements could be a single condition
if (measurement.name === 'measurements') || (measurement.name === 'scenarios-depth') ... etc

Or you could c onstruct an array of availble names like "eventOn = ['measurements', 'scenarios-depth' ...
Then use something like
if (eventOn.indexOf(measurement.name) >= 0) { addEventListen ....